packages/aleo-types/src/transition.rs

```diff
@@ -0,0 +1,67 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use error_stack::{Report, ResultExt};
+use serde::{Deserialize, Serialize};
+
+use crate::Error;
+
+const TRANSITION_PREFIX: &str = "au";
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Transition {
+    transition_id: String,
+}
+
+impl FromStr for Transition {
+    type Err = Report<Error>;
+
+    fn from_str(transition_id: &str) -> Result<Self, Self::Err>
+    where
+        Self: Sized,
+    {
+        crate::verify_bech32m(transition_id, TRANSITION_PREFIX)
+            .change_context(Error::InvalidAleoTransition(transition_id.to_string()))?;
```

Comment:
Now that you've simplified this function, I don't know that you even need the `Transition` and `Transaction` type. Why not just call this `verify_bech32m` function directly when parsing the message id? I'm sorry to want to minimize so much, it's just I want to avoid adding code for every new chain. I would like to keep chain specific code very minimal, else this repo will explode.